### PR TITLE
move plots to own folder with Project.toml, add PlotlyJS functionality

### PR DESCRIPTION
**Pull request by Lilyhanig**
- add PlotlyJS functionality
- move plots to own folder
- move plots test
- add a couple of tests

### CODE REVIEW

File: PowerSimulationsPlots/src/call_plots.jl [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+```
+
+# Accepted Key Words
+- `display::Bool`: set to false to prevent the plots from displaying
+- `save::String = "file_path"`: set a file path to save the plots
+- `seriescolor::Array`: Set different colors for the plots
+"""
+
+function bar_plot(res::PSI.Results; kwargs...)
+    backend = Plots.backend()
```
Comment (daniel-thom) [inline PowerSimulationsPlots/src/call_plots.jl]: Can we dispatch on the type of backend?

**Comment by daniel-thom** [comment]
Can you run JuliaFormatter on all files?

File: PowerSimulationsPlots/src/call_plots.jl [hunk 2]
```diff
@@ -193,47 +228,57 @@ function bar_plot(res::PSI.Results; kwargs...)
     save_fig = get(kwargs, :save, nothing)
```
Comment (daniel-thom) [inline PowerSimulationsPlots/src/call_plots.jl]:
```suggestion
    save_fig = get(kwargs, :save, true)
```

File: PowerSimulationsPlots/src/call_plots.jl [hunk 3]
```diff
@@ -193,47 +228,57 @@ function bar_plot(res::PSI.Results; kwargs...)
...
+    save_fig::Any,
+    set_display::Bool;
+    kwargs...,
+)
+    seriescolor = get(kwargs, :seriescolor, gr_default)
+    for name in string.(keys(res.variables))
+        variable_bar = get_bar_plot_data(res, name)
+        p = RecipesBase.plot(variable_bar, name; seriescolor = seriescolor)
+        set_display && display(p)
+        if !isnothing(save_fig)
```
Comment (daniel-thom) [inline PowerSimulationsPlots/src/call_plots.jl]: After the change above:
```suggestion
        if save_fig
```

And then there are several more instances of this.

File: PowerSimulationsPlots/src/call_plots.jl [hunk 4]
```diff
@@ -252,53 +297,44 @@ function stack_plot(res::PSI.Results; kwargs...)
...
-    stack_plot(results; kwargs...)
+function _stack_plot_internal(
+    res::PSI.Results,
+    stack::StackedGeneration,
+    backend::Plots.PlotlyJSBackend,
+    save_fig::Any,
+    set_display::Bool;
+    kwargs...,
+)
+    seriescolor = get(kwargs, :seriescolor, plotly_default)
```
Comment (daniel-thom) [inline PowerSimulationsPlots/src/call_plots.jl]: Can you make all global constants capitalized so that it's obvious what they are?  (PLOTLY_DEFAULT)

**Comment by jd-lara** [comment]
@Lilyhanig once we merge #317 you can rebase fix the conflicts and merge into dev0-2

**Pull request merged**